### PR TITLE
[1LP][RFR] Cluster Collection conversion

### PR DESCRIPTION
**Pull request by akarol**
Purpose or Intent
=================
Cluster collection conversion and delete test using collection

{{pytest: -v -k 'test_delete_cluster_from_table'}}

### CODE REVIEW

File: cfme/infrastructure/cluster.py [hunk 1]
```diff
@@ -132,6 +133,43 @@ def is_displayed(self):
             super(TimelinesView, self).is_displayed)
 
 
+class ClusterCollection(Navigatable):
```
Comment (quarckster) [inline cfme/infrastructure/cluster.py]: **REQURIED**
`ClusterCollection` should have `create()` and `instantiate()` methods. Look at Automate code as a reference.

File: cfme/infrastructure/cluster.py [hunk 2]
```diff
@@ -132,6 +133,43 @@ def is_displayed(self):
...
+                    checked_clusters.append(cluster)
+                    row[0].check()
+                    break
+            if set(clusters) == set(checked_clusters):
+                break
+        if set(clusters) != set(checked_clusters):
+            raise ValueError('Some Clusters were not found in the UI')
+        view.toolbar.configuration.item_select('Remove selected items', handle_alert=True)
+        view.entities.flash.assert_no_error()
+        flash_msg = \
```
Comment (quarckster) [inline cfme/infrastructure/cluster.py]: **REQUIRED**
Please use `()` to break lines.

File: cfme/infrastructure/cluster.py (hunk 1 above)
Comment (akarol) [inline cfme/infrastructure/cluster.py]: @quarckster We cannot add clusters from UI so create method will not exists.
Have added the instantiate method.

File: cfme/infrastructure/cluster.py (hunk 2 above)
Comment (akarol) [inline cfme/infrastructure/cluster.py]: Fixed flash msg

File: cfme/infrastructure/cluster.py [hunk 3]
```diff
@@ -185,18 +226,24 @@ def delete(self, cancel=True, wait=False):
...
+            self.provider.refresh_provider_relationships()
+            self.wait_for_disappear()
 
-            wait_for(lambda: not self.exists, fail_condition=False, num_sec=500, fail_func=refresh,
-                     message='Wait cluster to disappear')
+    def wait_for_disappear(self, timeout=300):
+        self.provider.refresh_provider_relationships()
+        try:
+            return wait_for(lambda: self.exists,
+                            fail_condition=True,
```
Comment (quarckster) [inline cfme/infrastructure/cluster.py]: **REQUIRED**
```python
return wait_for(lambda: not self.exists,
```
remove `fail_condition=True`

File: cfme/infrastructure/cluster.py [hunk 4]
```diff
@@ -185,18 +226,24 @@ def delete(self, cancel=True, wait=False):
...
+                            message='Wait for cluster to disappear',
+                            delay=10,
+                            fail_func=self.browser.refresh)
+        except TimedOutError:
+            logger.error('Timed out waiting for cluster to disappear, continuing')
 
     def wait_for_exists(self):
         """Wait for the cluster to be refreshed"""
-        view = navigate_to(self, 'All')
+        view = navigate_to(ClusterCollection, 'All')
```
Comment (quarckster) [inline cfme/infrastructure/cluster.py]: **REQUIRED**
```python
view = navigate_to(self.collection, 'All')
```

File: cfme/infrastructure/cluster.py [hunk 5]
```diff
@@ -221,7 +268,7 @@ def get_detail(self, *ident):
 
     @property
     def exists(self):
-        view = navigate_to(self, 'All')
+        view = navigate_to(ClusterCollection, 'All')
```
Comment (quarckster) [inline cfme/infrastructure/cluster.py]: **REQUIRED**
```python
view = navigate_to(self.collection, 'All')
```

File: cfme/infrastructure/cluster.py (hunk 1 above)
Comment (mfalesni) [reply cfme/infrastructure/cluster.py]: Yes, instantiate is a must, create depends :)

**Pull request merged**